subgraphs/aave-forks/src/mapping.ts

```diff
@@ -372,707 +513,584 @@ export function _handleReserveDataUpdated(
+    InterestRateType.STABLE
   );
 
-  const debtAsset = Token.load(debtToken.toHexString());
-  if (!debtAsset) {
-    log.warning("[Liquidate] Debt asset not found on protocol: {}", [
-      debtToken.toHexString(),
-    ]);
-    return;
+  const sBorrowerPositionBalance = sBorrowerPosition._getPositionBalance();
+  // Stable debt is liquidated after exhuasting variable debt
+  if (
+    debtBalances[1].equals(BIGINT_ZERO) &&
+    sBorrowerPositionBalance &&
+    sBorrowerPositionBalance.gt(BIGINT_ZERO)
+  ) {
+    sBorrowerPosition.subtractPosition(
+      event,
+      protocol,
+      debtBalances[0],
+      TransactionType.LIQUIDATE,
+      debtTokenMarket.inputTokenPriceUSD
+    );
+    liquidatedPositions.push(sBorrowerPosition.getPositionID()!);
   }
 
-  liquidate.position = positionId;
-  liquidate.blockNumber = event.block.number;
-  liquidate.timestamp = event.block.timestamp;
-  liquidate.liquidator = liquidator.toHexString();
-  liquidate.liquidatee = borrower.toHexString();
-  liquidate.market = market.id;
-  liquidate.hash = event.transaction.hash.toHexString();
-  liquidate.nonce = event.transaction.nonce;
-  liquidate.logIndex = event.logIndex.toI32();
-  liquidate.asset = debtAsset.id;
-  liquidate.amount = amount;
-  liquidate.amountUSD = amount
-    .toBigDecimal()
-    .div(exponentToBigDecimal(inputToken!.decimals))
-    .times(market.inputTokenPriceUSD);
-  liquidate.profitUSD = liquidate.amountUSD.minus(
-    debtToCover
-      .toBigDecimal()
-      .div(exponentToBigDecimal(debtAsset.decimals))
-      .times(repayTokenMarket.inputTokenPriceUSD)
-  );
+  liquidate.positions = liquidatedPositions;
   liquidate.save();
+}
 
-  protocol.cumulativeLiquidateUSD = protocol.cumulativeLiquidateUSD.plus(
-    liquidate.amountUSD
-  );
-  protocol.save();
-  market.cumulativeLiquidateUSD = market.cumulativeLiquidateUSD.plus(
-    liquidate.amountUSD
+export function _handleFlashLoan(
+  asset: Address,
+  amount: BigInt,
+  account: Address,
+  procotolData: ProtocolData,
+  event: ethereum.Event,
+  premiumAmount: BigInt,
+  flashloanPremium: _FlashLoanPremium
+): void {
+  const market = getMarketFromToken(asset, procotolData);
+  if (!market) {
+    log.warning("[_handleFlashLoan] market for token {} not found", [
+      asset.toHexString(),
+    ]);
+    return;
+  }
+  const manager = new DataManager(
+    market.id,
+    market.inputToken,
+    event,
+    procotolData
   );
-  market.save();
+  const tokenManager = new TokenManager(asset, event);
+  const amountUSD = tokenManager.getAmountUSD(amount);
+  const premiumUSDTotal = tokenManager.getAmountUSD(premiumAmount);
+  const flashloan = manager.createFlashloan(asset, account, amount, amountUSD);
+  flashloan.feeAmount = premiumAmount;
+  flashloan.feeAmountUSD = premiumUSDTotal;
+  flashloan.save();
+
+  let reserveFactor = market.reserveFactor;
+  if (!reserveFactor) {
+    reserveFactor = BIGDECIMAL_ZERO;
+  }
+  const protocolRevenueShare = reserveFactor;
```

Comment:
this should be https://etherscan.io/address/0x87870bca3f3fd6335c3f4ce8392d69350b4fa4e2#readProxyContract#F4

~ 4% rn which means premiumUSDToPortocol in the example transaction should be ~$2090

And PremiumToLP ~$50,169.6

Reply:
@dmelotik , thanks for helping looking to this. I don't think that's it. The totalRevenueDelta at the ReserveDataUpdate event in my example tx is about $62, way lower than $50,169 and we will still get a very negative revenue.